include/Board.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include "Piece.h"
+#include "macros.h"
+#include "Factory.h"
+#include <memory>
+#include <iostream>
+
+
+
+class Board {
+public:
+	Board(const std::string& board);
+	int getMove(const std::string& move);
+
+private:
+	void initBoard();
+	bool isEmpty(const Location) const;
+	bool isRival(const Location) const;
+	bool isCurrentPiecePlayer(const Location) const;
+	bool isWayBlocked(const Location&, const Location&) const;
+	bool moveCausesChess(const Location&, const Location&);
```

Comment:
The Board class seems to have multiple responsibilities, such as maintaining the game state and handling move operations. Consider separating these responsibilities into different classes or functions to adhere to the Single Responsibility Principle.